staging/src/k8s.io/client-go/tools/cache/fifo.go

```diff
@@ -184,14 +172,7 @@ func (f *FIFO) Add(obj interface{}) error {
 // safely retry items without contending with the producer and potentially enqueueing
 // stale items.
 func (f *FIFO) AddIfNotPresent(obj interface{}) error {
-	id, err := f.keyFunc(obj)
-	if err != nil {
-		return KeyError{obj, err}
-	}
-	f.lock.Lock()
-	defer f.lock.Unlock()
-	f.addIfNotPresent(id, obj)
-	return nil
+	return f.addItem(obj, true)
```

Comment:
If renaming meaning of second arg to `shouldReplace`, then this `true` flips to `false. Same with the other function